app/store_loader_test.go

```diff
@@ -0,0 +1,129 @@
+package app
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/store/rootmulti"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+)
+
+func TestWrapStoreLoader(t *testing.T) {
+	var flag bool
+	tests := []struct {
+		name        string
+		storeLoader baseapp.StoreLoader
+		wrapper     StoreLoaderWrapper
+		err         string
+	}{
+		{
+			name:        "nil store loader is set with valid value",
+			storeLoader: nil,
+			wrapper:     createMockStoreWrapper(&flag),
+		},
+		{
+			name:        "nil wrapper is handled",
+			storeLoader: createMockStoreLoader(),
+			wrapper:     nil,
+			err:         "wrapper must not be nil",
+		},
+		{
+			name:        "contents of wrapper are called",
+			storeLoader: createMockStoreLoader(),
+			wrapper:     createMockFlipWrapper(&flag),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			storeLoader := WrapStoreLoader(tc.wrapper, tc.storeLoader)
+			db := dbm.GoLevelDB{}
+			ms := rootmulti.NewStore(&db, nil)
```

Comment:
When creating these test dbs/stores, you need to use a temporary directory provided by `t.TempDir()`. If you intend to re-use the same db/store across multiple tests you'd use the `t` outside this `t.Run(...)`. Alternatively, maybe use a `MemDB`. I'm pretty sure that doesn't create any dirs/files on the system.

The problem is that it ends up using some default location which is sometimes even the default `--home` used by `provenanced` which can then mess other things up on the system running the tests. It can also cause tests to fail because that already exists but is not what the test is expecting. Judging by the `app/.db` dir in this PR, I'm guessing it's using a default of the repo's root directory since that